pinot-spi/src/main/java/org/apache/pinot/spi/config/table/UpsertConfig.java

```diff
@@ -189,6 +197,12 @@ public void setDeleteRecordColumn(String deleteRecordColumn) {
     }
   }
 
+  public void setOutOfOrderRecordColumn(String outOfOrderRecordColumn) {
+    if (outOfOrderRecordColumn != null) {
+      _outOfOrderRecordColumn = outOfOrderRecordColumn;
+    }
```

Comment:
We should not have this `null` check`. Can you also remove the `null` check for `deleteRecordColumn`? It is needed only for `comparisonColumn` to solve the backward compatible issue introduced when adding `comparisonColumns`
```suggestion
    _outOfOrderRecordColumn = outOfOrderRecordColumn;
```